### PR TITLE
Issue 14: Add initial FilesXml and Metadata classes

### LINKED ISSUE #14: FileItem structuur
Implement a data structure (`FilesXml`) that can fully represent `files.xml`.

Here is a small starter already to which other data structures are added in the constructor. Leave the unimplemented methods unimplemented for now.

```scala
package nl.knaw.dans.bag.v0

import java.io.InputStream

import better.files.File

import scala.util.Try

case class FilesXml() {

  def save(file: File): Try[Unit] = ???
}

object FilesXml {

  def empty(): FilesXml = {
    FilesXml()
  }

  def read(file: File): Try[FilesXml] = ???

  def read(inputStream: InputStream): Try[FilesXml] = ???
}
```

Add this to a (to be made) case class `Metadata`:

```scala
package nl.knaw.dans.bag.v0

import better.files.File
import nl.knaw.dans.bag.v0.Metadata.filesXmlFilename

import scala.util.Try

case class Metadata private(filesXml: FilesXml) {

  def save(dir: File): Try[Unit] = {
    for {
      _ <- filesXml.save(dir / filesXmlFilename)
    } yield ()
  }
}

object Metadata {

  val filesXmlFilename = "files.xml"

  def empty(): Metadata = {
    Metadata(FilesXml.empty())
  }

  def read(dir: File): Try[Metadata] = {
    for {
      filesXml <- FilesXml.read(dir / filesXmlFilename)
    } yield Metadata(filesXml)
  }
}
```

### PR DESCRIPTION
**Pull request by lindareijnhoudt**
fixes #14 

#### When applied it will
* Define the FilesXml, FilesXmlItem and Metadata classes
* Provide a listing of all dc/dcterms elements allowed in a FilesXmlItem

#### Where should the reviewer @DANS-KNAW/easy start?

* [ ] vinkje voor Jan

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/bag/v0/FilesXml.scala [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  val ACCESS_RIGHTS: DcTermsElementType = Value("accessRights")
+  val ACCRUAL_METHOD: DcTermsElementType = Value("accrualMethod")
+  val ACCRUAL_PERIODICITY: DcTermsElementType = Value("accrualPeriodicity")
+  val ACCRUAL_POLICY: DcTermsElementType = Value("accrualPolicy")
+  val ALTERNATIVE: DcTermsElementType = Value
+  val AUDIENCE: DcTermsElementType = Value
+  val AVAILABLE: DcTermsElementType = Value
+  val BIBLIOGRAPHIC_CITATION: DcTermsElementType = Value("bibliographicCitation")
+  val CONFORMS_TO: DcTermsElementType = Value("conformsTo")
+//  val CONTRIBUTOR: DcTermsElementType = Value
```
Comment (lindareijnhoudt) [inline src/main/scala/nl/knaw/dans/bag/v0/FilesXml.scala]: these are already defined in the dc-realm.

File: src/main/scala/nl/knaw/dans/bag/v0/FilesXml.scala [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+object XmlLang extends Enumeration {
+  type XmlLang = Value
+
+  val ENG: XmlLang = Value
+  val NLD: XmlLang = Value
+  val FRA: XmlLang = Value
+  val DEU: XmlLang = Value
+}
+
+object DcTermsElementType extends Enumeration {
```
Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/bag/v0/FilesXml.scala]: This is not limited to FilesXml but also applies to DatasestXml so put it in a separate file. It is not even limited to V0 so put it in another package.

File: src/main/scala/nl/knaw/dans/bag/util/constants.scala [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  val BIBLIOGRAPHIC_CITATION: DcTermsElementType = Value("bibliographicCitation")
+  val CONFORMS_TO: DcTermsElementType = Value("conformsTo")
+  //  val CONTRIBUTOR: DcTermsElementType = Value
+  //  val COVERAGE: DcTermsElementType = Value
+  val CREATED: DcTermsElementType = Value
+  //  val CREATOR: DcTermsElementType = Value
+  //  val DATE: DcTermsElementType = Value
+  val DATE_ACCEPTED: DcTermsElementType = Value("dateAccepted")
+  val DATE_COPYRIGHTED: DcTermsElementType = Value("dateCopyrighted")
+  val DATE_SUBMITTED: DcTermsElementType = Value("dateSubmitted")
```
Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/bag/util/constants.scala]: Not even considering the overall approach, this is a too big unstructured pile of types.
How to reuse (for example) dates by the [deposit-api](https://github.com/DANS-KNAW/easy-deposit-api/blob/39bfa9fda60354640146f6a517162246a4f470f3/src/main/scala/nl.knaw.dans.easy.deposit/docs/dm/Date.scala#L24-L35) other enums in the same dm package of the api, relation types and whatever?

**Comment by rvanheest** [comment]
This has been open for more than a year now. We still need to decide how to implement this, but so far we haven't done anything with it. Closing this PR for now.

**Pull request closed without merging**